multiversx_sdk/entrypoints/entrypoints.py

```diff
@@ -118,45 +141,81 @@ def get_transaction(self, tx_hash: Union[str, bytes]) -> TransactionOnNetwork:
 
     def create_network_provider(
         self,
-    ) -> Union[ApiNetworkProvider, ProxyNetworkProvider]:
+    ) -> INetworkProvider:
         return self.network_provider
 
     def create_delegation_controller(self) -> DelegationController:
+        if not self.chain_id:
```

Comment:
Can we extract this to a private method e.g. `_acquire_chain_id_if_missing()` or something like that?

We could have also fetched it in the constructor (a single place in the code), but then we would had broken the rule of "don't touch remote systems in class constructors".

Reply:
extracted in a method